general/forall.hpp

```diff
@@ -32,6 +32,21 @@ const int MAX_D1D = 14;
 const int MAX_Q1D = 14;
 #endif
 
+/** @brief Create a scratch memory on the device. */
```

Comment:
While you mention this is intended  for the device, it seems that it would still be possible to use this on the host for non-device builds. Due to that fact, if a user was using an openmp version of mfem and for some reason was calling this from a threaded loop it seems like you could run into some race-conditions here. So, it might not hurt to add a note in the function description that this is not thread-safe.

Reply:
On another note, it seems once allocated this memory will always either stay the same size or grow. Would it be worth adding in a variable to allow for the capacity to shrink if a user knows it will never reach a maximum size again in order to regain some of that scratch memory back?

Reply:
I'll add the note and one method to erase the scratch memory.

Reply:
Scratch(pad) memory is an internal memory used for temporary storage to hold small items of data.

Reply:
Removed - Will be addressed in another PR.